vizro-ai/docs/pages/user-guides/advanced-options.md

```diff
@@ -0,0 +1,205 @@
+# How to use Vizro-AIs advanced options
+
+This guide shows you how to use the advanced options of `vizro_ai.plot`.
+
+First we show how to change the input parameters of the function, allowing the control of whether code gets executed, the number of retries of `.plot` when
+it fails validation, and how to change for a more comprehensive return of output (when `return_elements=True`).
+
+Second we show how to use this more comprehensive output, allowing the control of code generation and fig object production.
+
+## Inputs of `vizro_ai.plot`
+
+### `user_input`
+This is the natural language query from which, together with a data sample, the LLM creates a plotly chart. The complexity of charts possible is dependent [on the different vendor model capabilites](customize-vizro-ai.md#what-model-to-choose). It is also [possible to use different languages](use-different-languages.md).
+
+
+### `df`
+Provide any `pandas` data frame to base your query on. The LLM will receive a sample of this data frame to form an appropriate graph. 
+
+If the option `validate_code` is set to `True` (which it is by default), the LLM created chart code will be evaluated on a sample of this data frame.
+
+If `return_elements` is set to `False`, then the returned fig object will be created based on this (entire) data frame.
+
+### `max_debug_retry`
+This number determines how often the tool will try to correct an incorrect response (that fails various validation criteria). Under the hood this is [implemented via pydantic validators](https://docs.pydantic.dev/1.10/usage/validators/). The last response will be re-sent to the LLM together with the validation error(s) in order to receive an improved response. This concept is [inspired by the amazing instructor library](https://github.com/jxnl/instructor).
+
+### `return_elements`
+This boolean (by default `False`) determines the return type of `vizro_ai.plot`. 
+
+If set to `False`, then dynamically generated python code is executed in order to produce a `plotly.graph_objects.Figure` object from the LLM response and the user provided data frame. Strictly speaking, it produces a `vizro.charts._charts_utils._DashboardReadyFigure`, which behaves essentially like a plotly graph object, but is ready to be [inserted](add-generated-chart-usecase.md) into a [Vizro](https://vizro.readthedocs.io/en/stable/) dashboard. It also comes with the default Vizro dark theme.
```

Comment:
Shall we not expose "_DashboardReadyFigure" to users? especially in how to guides

Reply:
You mean not mention it? Yh....

Reply:
yes, `_DashboardReadyFigure ` is meant to be private no?

Reply:
So I thought about this for a while, I feel like we can leave it in. The reason being that this is an advanced guide, and we make it amply clear that this is only important if someone wants to know the _exact_ return type. It quickly says how it behaves in general